modules/runner/src/main/java/org/apache/ignite/runner/internal/app/IgnitionImpl.java

```diff
@@ -79,15 +81,21 @@
 
         boolean cfgBootstrappedFromPds = vaultMgr.bootstrapped();
 
-        // Bootstrap local configuration manager.
-        ConfigurationManager locConfigurationMgr = new LocalConfigurationManagerImpl();
+        List<RootKey<?, ?>> rootKeys = new ArrayList<>(Collections.singletonList(NetworkConfiguration.KEY));
 
-        locConfigurationMgr.configurationRegistry().registerRootKey(NetworkConfiguration.KEY);
+        List<ConfigurationStorage> configurationStorages =
+            new ArrayList<>(Collections.singletonList(new LocalConfigurationStorage(vaultMgr)));
 
-        locConfigurationMgr.configurationRegistry().registerStorage(new LocalConfigurationStorage(vaultMgr));
+        // Bootstrap local configuration manager.
+        ConfigurationManager locConfigurationMgr = new ConfigurationManager(rootKeys, configurationStorages);
 
         if (!cfgBootstrappedFromPds)
-            locConfigurationMgr.bootstrap(jsonStrBootstrapCfg);
+            try {
+                locConfigurationMgr.bootstrap(jsonStrBootstrapCfg);
```

Comment:
I think need to create a Network manager, to move all code about starting a network toward.